Versioning_oM/Attributes/VersioningTargetAttribute.cs

```diff
@@ -0,0 +1,63 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2025, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using BH.oM.Base.Debugging;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace BH.oM.Versioning
+{
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+    [Description("Attribute defining the specfic type or method a versioning method is targetting. This attribute can be used multiple times if a method is targetting more than one element.")]
```

Comment:
Don't think this is used for methods? I have at least never seen it used for methods, and looking at the `UpgradeMethod` is has no references to the `ToNewObject` that stores the upgrade methods. Guess we could change that if we like, but given the `PreviousVersionAttributes` already works well, I see no real need.

Also, added a minor comment about that the type name need to be fully namespace qualified.

```suggestion
    [Description("Attribute defining the specific type a versioning method is targeting. The target type needs to be specified including the full namespace. This attribute can be used multiple times if a method is targeting more than one element. ")]
```